### PR TITLE
feat: update cloud provider rate limit for vmas.

### LINKED ISSUE #1806: enable smart cloudprovider rate limiting for VMAS agent node deployment.
Kubernetes version:
1.13
What happened:
The logs for kube-manager show frequent errors for being rate limited on Azure Stack kubernetes cluster deployment when we try to deploy an application with service type load balancer.

### PR DESCRIPTION
**Pull request by rjaini**
**Reason for Change**:
PR changes the cloudprovider-enforced rate limiter configuration for VMAS scenarios as well to allow for more "burstability". For large cluster on Azure Stack, app deployment which requires LoadBalancer reconciliation either NicGet call or NicUpdate call is failing. Updating the code to support "burstability" on VMAS as well.

**Issue Fixed**:
Fixes #1806 

**Requirements**:
- [X] uses [conventional commit messages](https://www.conventionalcommits.org/) 
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/api/types.go [hunk 1]
```diff
@@ -1834,15 +1834,8 @@ func (p *Properties) IsNvidiaDevicePluginCapable() bool {
...
-		if p.HasVMSSAgentPool() && !p.IsHostedMasterProfile() {
-			var rateLimitBucket int
-			for _, profile := range p.AgentPoolProfiles {
-				if profile.AvailabilityProfile == VirtualMachineScaleSets {
-					rateLimitBucket += common.MaxAgentCount
-				}
-
-			}
-			p.OrchestratorProfile.KubernetesConfig.CloudProviderRateLimitBucket = rateLimitBucket
+		if p.AnyAgentUsesAvailabilitySets() || p.AnyAgentUsesVirtualMachineScaleSets() {
```
Comment (jackfrancis) [inline pkg/api/types.go]: Thanks for simplifying the per-agent pool multiplier :)

Let's just get rid of the `if p.AnyAgentUsesAvailabilitySets() || p.AnyAgentUsesVirtualMachineScaleSets() {` altogether as there's no other kind of agent pool. We can also retire the `DefaultKubernetesCloudProviderRateLimitBucket` const if it's no longer being used.

Comment (rjaini) [reply pkg/api/types.go]: @jackfrancis : Sure we can do that. Only reason I didn't remove as was not sure if empty AgentPoolProfiles supported (load to default 3 Linux node deployment) ?.

Comment (jackfrancis) [reply pkg/api/types.go]: Perhaps we can accommodate that by keeping the default const in the event that we have zero pools (and thus try to set the rate to zero)

Comment (rjaini) [reply pkg/api/types.go]: I didn't get what you meant above.
Is below snippet what you are looking for where we set default value when agent poll is count is 0 ?.
		var agentPoolProfileCount int
		agentPoolProfileCount = len(p.AgentPoolProfiles)
		if agentPoolProfileCount == 0 {
			p.OrchestratorProfile.KubernetesConfig.CloudProviderRateLimitBucket = DefaultKubernetesCloudProviderRateLimitBucket
		} else {
			p.OrchestratorProfile.KubernetesConfig.CloudProviderRateLimitBucket = agentPoolProfileCount * common.MaxAgentCount

		}

Comment (jackfrancis) [reply pkg/api/types.go]: yes

Comment (rjaini) [reply pkg/api/types.go]: Updated the PR with the changes.

**Pull request merged**